WRITING TESTS.md

```diff
@@ -0,0 +1,141 @@
+# Writing and running tests on Deriv App
+
+This document is intended to be a guideline for writing and running tests on deriv app.
+
+Things that this document covers:
+
+* Brief explanation on test framework implemented on the project.(Jest)
+
+* We will describe how `react-testing-library` configured. 
+
+* An example of how to write test on using implemented framework
+
+   
+
+  ## Test Framework explanation 
+
+  ​	We are using `Jest` as test runner and [React Testing library](https://testing-library.com/docs/react-testing-library/intro/) as our main test component to have the ultimate needed utilities.
+
+  
+
+  ### 	Jest 
+
+  ​		With Jest we can run our tests quickly and even parallel.
+
+  ​		All of needed configurations are located in root folder of the app.
+
+  
+
+  #### 	Run Tests
+
+  ​		To Run tests from root directory simply run:
+
+  ​			```npm run test:jest```
+
+  ​		Be informed that this command will run all of the tests in the whole app based on following configuration provided.
+
+  ​		Also the mentioned command will generate a coverage directory in the root directory, you can check the current coverage percentage by the generated html.
+
+  
+
+   
+
+  #### 	Jest configuration files explanation
+
+  ​	
+
+  * `jest.config.js`
+  
+    General configuration of the jest which will use to run the tests from root directory of deriv-app 
+  
+    1. `collectCoverage`: with this flag jest will collect the coverage report in order to know how much and which files of the code has test coverage.
+    2. `collectCoverageFrom`: To indicate which files should and should not test. Generally we exclude all of `node_modules` directories.
+    3. `collectCoverageFrom`: The type of needed coverage report.
+    4. `clearMocks`: configuration option to clear mocks automatically before each tests.   
+    5. `projects`: To indicate that which components/directories should test.
+    6. `transform`: A map from regular expressions to paths to transformers. A transformer is a module that provides a synchronous function for transforming source files.
+    7. `testRegex`: The pattern or patterns Jest uses to detect test files.
+    8. `transformIgnorePatterns`: An array of regexp pattern strings that are matched against all source file paths before transformation. If the file path matches **any** of the patterns, it will not be transformed.
+
+  
+
+  * `jest.config.base.js`
+
+    This Configuration file is held in common across all of the individual packages. In other words if you want to just run tests related to one package (eg. trader) this is what that particular package will use as base configuration.
```

Comment:
```suggestion
    This Configuration file exists in all of the individual packages. In other words if you want to just run tests related to one package (e.g. trader) this is what that particular package will use as base configuration.
```

Reply:
It doesn't exist in all of the packages, We just have it in the root directory of the project and we are extending it in the internal packages.

Reply:
It doesn't exist in all of the packages, We just have it in the root directory of the project and we are extending it in the internal packages

Reply:
Cool :+1: